docs/recipes/endpoint-testing-with-mongoose.md

```diff
@@ -0,0 +1,281 @@
+# Endpoint testing with Mongoose
+
+This recipe shows you how to test your endpoints with AVA and Mongoose.
+
+## Setup
+
+This recipe uses the following libraries:
+
+1. [MongoDB memory server](https://github.com/nodkz/mongodb-memory-server) (A MongoDB in-memory Server)
+2. [Babel Polyfill](https://babeljs.io/docs/usage/polyfill/) (required for MongoDB memory server)
+3. [Supertest](https://github.com/visionmedia/supertest) (An endpoint testing library)
+4. Mongoose
+
+Install the first three libraries by running the following code:
+
+```console
+$ npm install --save-dev mongodb-memory-server babel-polyfill supertest
+```
+
+You should have Mongoose installed already. If not, run the following code to install it:
+
+(Note: You need at least mongoose v4.11.3)
+
+```console
+$ npm install mongoose
+```
+
+## Setting up AVA
+
+Since MongoDB Memory server requires Babel polyfill to work, the easiest way to set up AVA is through the `ava` key in your `package.json` file.
```

Comment:
I find this misleading. It just needs `regenerator-runtime`, right? Going by https://www.npmjs.com/package/regenerator-runtime (which is depended on by `babel-polyfill`) you could install that as a dependency and then require `regenerator-runtime/runtime`. That should only be done if you're testing on Node.js releases older than 8.

`babel-polyfill` impacts your environment in many other ways. This recipe should focus on what is absolutely necessary.